bonus-kde.qmd

```diff
@@ -1,180 +1,231 @@
+-   Least Square Cross Validation (LSCV), which minimizes the difference in volume between the true UD and the estimates UD.
+-   A subjective visual choice for the smoothing parameter, based on successive trials (Silverman, 1986; Wand & Jones 1995).
 
-Hpi1 <- Hpi(x=x)
-Hpi2 <- Hpi.diag(x = x)
+Other parameters include:
 
+-   Kernel Type: The type of kernel is limited to Gaussian (bivariate normal), quadratic or normal. We used the bivariate normal model as default.
 
-fhat.pi1 <-kde(x=x,H=Hpi1) 
-fhat.pi2 <-kde(x=x,H=Hpi2)
+-   Grid/extent: These determines the area or extent over which the home range will be estimated. This is a mix of fine scale and time consuming processing and faster blocky resolution over a continuous surface. As a rule of thumb you GME suggests: take the square root of the x or y variance value (whichever is smaller) and divide by 5 or 10 (I usually round to the nearest big number - so 36.7 becomes 40). Before using this rule of thumb value calculate how many cells this will result in for the output (take the width and height of you input points, divide by the cell size, and multiply the resulting numbers together). If you get a value somewhere between 1-20 million, then you have a reasonable value.
 
+To create home ranges for each unique id we used a bivariate normal kernel with a variety of h (smoothing parameters). We then interpreted visually to determine size to use based on successive trials. This is supported by the literature (Hemson et al. 2005; Calenge et al. 2011).
 
-plot(fhat.pi1)
-plot(fhat.pi1)
+It is also possible to run KDE with set barriers and boundaries.
 
+#### 3.1 kde: h reference parameter
 
+```{r}
+  tdfu <- tdf  |>  dplyr::select(location.long, location.lat, season)
+    
+  # Create a SpatialPointsDataFrame by defining the coordinates
+  coordinates(tdfu) <- c("location.long", "location.lat")
+  proj4string(tdfu) <- CRS( "+proj=longlat +datum=WGS84 +units=m +no_defs" )
+  tdfgeo <- spTransform(tdfu, CRS("+init=epsg:3005")) # Transform to UTM 
+
+  # define the parameters (h, kern, grid, extent) 
+  kde_href  <- kernelUD(tdfgeo, h = "href", kern = c("bivnorm"), grid = 500, extent = 2)
+  kde_href
 ```
 
-# unconstrained Smoothed Cross Validation 
+From this object (**Utilization distribution of several Animals**) we can extract the vertices or polygons which define the percentage we wish to include.
 
 ```{r}
+  ver95 <- getverticeshr(kde_href,95) # get vertices for home range
+  ver95_sf <- st_as_sf(ver95)         # convert to sf object 
+  
+  ver75 <- getverticeshr(kde_href,75)
+  ver75_sf <- st_as_sf(ver75 )
+  
+  ver50 <- getverticeshr(kde_href,50)
+  ver50_sf<- st_as_sf(ver50)
+ 
+  # plot the outputs 
+  mapview(ver50_sf, zcol = "id") 
+  mapview (ver75_sf, zcol = "id")
+  mapview (ver95_sf, zcol = "id")
+  
+  plot(st_geometry(ver95.2.sf),col = "yellow") 
+  plot(st_geometry(ver75.2.sf),col = "blue", add = TRUE)
+  plot(st_geometry(ver50.2.sf),col = "purple", add = TRUE)
```

Comment:
Objects names `.2.sf` don't exist, I think you mean these?

```suggestion
  plot(st_geometry(ver95_sf),col = "yellow") 
  plot(st_geometry(ver75_sf),col = "blue", add = TRUE)
  plot(st_geometry(ver50_sf),col = "purple", add = TRUE)
```

Reply:
Thanks for catching that